File: CS/ReplicateBuiltInToolbar/ViewModels/BaseViewModel.cs
using DevExpress.Maui.Core;
using DevExpress.Maui.HtmlEditor;
using Microsoft.Maui.Controls;

namespace HtmlEditToolbarCustomization;

public abstract class BaseViewModel : BindableBase, IHtmlOwner {
    protected BaseViewModel(HtmlEdit owner) {
        Owner = owner;
    }

    public HtmlEdit? Owner {
        get => GetValue<HtmlEdit>();
        set => SetValue(value, (ov, nv) => {
            if (ov != null)
                OnDetachHtmlEdit(ov);
            if (nv != null)
                OnAttachHtmlEdit(nv);
        });
    }

    protected virtual void OnAttachHtmlEdit(HtmlEdit htmlEdit) {
    }

    protected virtual void OnDetachHtmlEdit(HtmlEdit htmlEdit) {
    }
}

File: VB/ReplicateBuiltInToolbar/ViewModels/BaseViewModel.vb
Imports DevExpress.Maui.Core
Imports DevExpress.Maui.HtmlEditor
Imports Microsoft.Maui.Controls

Namespace HtmlEditToolbarCustomization

    Public MustInherit Class BaseViewModel
        Inherits BindableBase
        Implements IHtmlOwner

        Protected Sub New(ByVal owner As HtmlEdit)
            Me.Owner = owner
        End Sub

        Public Property Owner As HtmlEdit? Implements IHtmlOwner.Owner
            Get
                Return GetValue(Of HtmlEdit)()
            End Get

            Set(ByVal value As HtmlEdit?)
                SetValue(value, Function(ov, nv)
                    If ov IsNot Nothing Then OnDetachHtmlEdit(ov)
                    If nv IsNot Nothing Then OnAttachHtmlEdit(nv)
                End Function)
            End Set
        End Property

        Protected Overridable Sub OnAttachHtmlEdit(ByVal htmlEdit As HtmlEdit)
        End Sub

        Protected Overridable Sub OnDetachHtmlEdit(ByVal htmlEdit As HtmlEdit)
        End Sub
    End Class
End Namespace
